src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,137 @@
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+import time
+from typing import Dict, FrozenSet, List, Optional, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.settings import CFG
+from predicators.src.structs import GroundAtom, Object, ParameterizedOption, \
+    Sequence, State, Task, _GroundNSRT
+
+
+class LLMProbeApproach(OpenLoopLLMApproach):
+    """LLMProbeApproach definition."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "llm_probe"
+
+    def _get_llm_based_plan(
+            self, state: State, atoms: Set[GroundAtom],
+            goal: Set[GroundAtom]) -> Optional[List[_GroundNSRT]]:
+        """Query the LLM and use the outputs to initialize the queue in A*."""
+        # To initialize the queue in A*, we will build up a partial policy
+        # and then run policy-guided A*.
+        start_time = time.time()
+        objects = set(state)
+        partial_policy_dict: Dict[FrozenSet[GroundAtom], _GroundNSRT] = {}
+        for option_plan in self._get_llm_based_option_plans(
+                atoms, objects, goal):
+            ground_nsrt_plan = self._option_plan_to_nsrt_plan(
+                option_plan, atoms, objects, goal)
+            assert ground_nsrt_plan is not None
+            # Simulate the ground NSRT plan and record the (atoms, ground NSRT)
+            # pairs encountered along the way to build up a partial policy.
+            single_partial_policy_dict = {}
+            cur_atoms = atoms
+            for ground_nsrt in ground_nsrt_plan:
+                # If we run into an invalid ground NSRT, the rest of the plan
+                # is going to be garbage, so stop here.
+                if not ground_nsrt.preconditions.issubset(cur_atoms):
+                    break
+                single_partial_policy_dict[frozenset(cur_atoms)] = ground_nsrt
+                cur_atoms = utils.apply_operator(ground_nsrt, cur_atoms)
+            # Update the overall partial policy. Note that there may be
+            # entries that get overridden. This is okay, and there's no
+            # reason to prefer one option plan over another, so we just let
+            # the overriding happen.
+            partial_policy_dict.update(single_partial_policy_dict)
+        # Run policy-guided A*. Note that we importantly want to use
+        # sesame_plan because we will want to report the metrics collected in
+        # sesame_plan.
+        abstract_policy = lambda a, o, g: partial_policy_dict.get(
+            frozenset(a), None)
+        max_policy_guided_rollout = CFG.horizon  # not important
```

Comment:
wait..why is this not important? is it because your abstract policy only "works" on a few states?

Reply:
yeah basically. adding a better comment